src/main/java/com/pinterest/secor/common/KafkaClient.java

```diff
@@ -143,7 +148,7 @@ private Message getMessage(TopicPartition topicPartition, long offset,
         }
         Long timestamp = null;
         if (mConfig.useKafkaTimestamp()) {
-            timestamp = messageAndOffset.message().timestamp();
+            timestamp = mKafkaMessageTimestampFactory.create(mKafkaTimestampClassName).getTimestamp(messageAndOffset);
```

Comment:
Instead of calling factory.create(className) each time, can the factory have the timestampClass created in its constructor so you just need to get that class each time?  Use reflection to create the java class each time is expensive.